cinema/models.py

```diff
@@ -125,22 +166,94 @@ class Meta:
         ordering = ["name"]
 
 
+class CinemaPersonManager(models.Manager):
+    """
+    Custom Manager, adding extra manager methods.
+
+    Custom Manager used in 'CinemaPerson' model by extending base Manager class
+    and instantiating custom Manager in 'CinemaPerson' model.
+    """
+    def get_queryset(self):
+        """
+        Return QuerySet object with cached data of some related models.
+        """
+        return super().get_queryset().select_related("user", "country")
+
+    def get_brief_persons_data(self):
+        """
+        Return QuerySet object with data of specified fields of
+        'CinemaPerson' and 'User' models.
+        """
+        return super().get_queryset().select_related("user").values_list(
+            "pk", "user__first_name",  "user__last_name", named=True
+        )
+
+    def filter_by_search_word(self, search_word):
+        """
+        Return QuerySet object, value of selected field corresponds to
+        search word entered by visitor.
+        """
+        split_query = search_word.split()
+        q = (Q(user__first_name__istartswith=split_query[0]) &
+             Q(user__last_name__iendswith=split_query[-1])) | Q(
+            user__first_name__icontains=search_word) | Q(
+            user__last_name__icontains=search_word
+        )
+        return self.get_queryset().filter(q)
+
+    def get_cached_persons_data(self):
+        """
+        Return QuerySet object with cached data of all related models.
+        """
+        return self.get_queryset().prefetch_related(
+            "news_set", "film_set__genre",
+            Prefetch("film_set", Film.objects.select_related("imdb_rating"))
+        )
+
+    def get_specified_persons_data(self, fields_names):
```

Comment:
`get_specified_persons_data` is really bad name. The name says nothing. Please always think hard to choose any name wisely.

Reply:
ok